File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int count = nums.length/2;
        
        nums = Arrays.stream(nums).distinct().toArray();
        int count1 = nums.length;

        if(count1 < count){
            answer = count1;
        }else {
            answer = count;
        }

        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    count = len(nums) // 2  # 나눗셈 결과를 정수로 변환

    unique_nums = set()  # set 객체를 생성

    for i in nums:
        unique_nums.add(i)  # set에 요소를 추가
    
    if len(unique_nums) < count:
        answer = len(unique_nums)
    else:
        answer = count

    return answer
